src/components/Button/index.tsx

```diff
@@ -37,6 +39,9 @@ export type ButtonProps = {
   size: 'sm' | 'md' | 'lg' | 'xl';
   title: string;
   disabled?: boolean;
+  icon?: string;
+  iconPosition?: 'left' | 'right';
+  iconSize?: number; // Default will follows button's fontSize but also dynamically applicable
```

Comment:
ps: kita bisa bikin comment buat props dibaca pas di hover dari vscode, contohnya

<img width="320" alt="image" src="https://user-images.githubusercontent.com/23289654/166889239-f2537ec6-9588-4d3c-9c8b-bece635d256b.png">

caranya jadiin commentnya sebelum props, terus commentnya pake `/**  comment here */`, e.g.


```suggestion
  /** Default will follows button's fontSize but also dynamically applicable */
  iconSize?: number
```

Reply:
mantab, terimakasih TS god